src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -57,12 +130,26 @@ internal class LinuxAppServiceLogger
         {
             // Initializing fixed logging metadata
             this.writeToConsole = writeToConsole;
-            this.roleInstance = JToken.FromObject("App-" + containerName);
-            this.tenant = JToken.FromObject(tenant);
-            this.stamp = stampName;
 
-            var finalCharIndex = stampName.Length - 1;
-            this.primaryStamp = char.IsLetter(stampName[finalCharIndex]) ? stampName.Remove(finalCharIndex) : stampName;
+            // Since the values below are obtained via a NameResolver, they might be null.
+            // Attempting to serialize a null value results in exceptions, or even worse, wrong logs,
+            // so we need to be careful.
+            if (!string.IsNullOrEmpty(containerName))
+            {
+                this.roleInstance = "App-" + containerName;
+            }
+
+            if (!string.IsNullOrEmpty(tenant))
```

Comment:
What's the purpose of this `if` check? I'm not noticing any particular behavior different vs. just always assigning `this.tenant = tenant;`.

Reply:
Right, I did notice this but had not made the change yet as I was hoping this push would go un-noticed. That's a bad assumption on my part, that's why I made this back into a draft. In any case, I'm pushing a new commit with explanations for all of this and responding to your feedback